Cython/Compiler/Symtab.py

```diff
@@ -588,7 +588,10 @@ def declare_type(self, name, type, pos,
         if defining:
             self.type_entries.append(entry)
 
-        if not template:
+        # DW - TODO there's potentially a lot of cases where we replace the
+        # entry of a type and we shouldn't do. I'm suspicious of redeclaring
+        # entries in derived cppclasses for example.
+        if not (template or dont_set_entry):
             type.entry = entry
```

Comment:
Should we maybe check if a `.type` is already set and not overwrite it?
As you may remember from past cases, I'm generally wary about newly adding options to fix special cases. Special cases are rarely special enough to break the rules.

Reply:
When I was investigating I spotted one very frequent case where `type` is re-assigned that does look intentional: `CythonScope` is recreated every time a `Context` is created and the fused types (`cython.integral` etc) have their entries re-assigned.

However, I suspect that this can't matter (much?) because it'd break parallel builds if it did.

The other option I considered was to rename the `template` argument since all `template` does is prevent the entry from being set.

So yeah - I mostly agree here, but it needs a bit more investigation to work out what simplification works

Reply:
It looks like "don't overwrite `.entry` if set" works, so I'll go with that